spec/page_client_spec.rb

```diff
@@ -53,14 +53,15 @@
     pages = [Page.new(filename: filename_1, structure: structure_1, position: 1),
              Page.new(filename: filename_2, structure: structure_2, position: 2)]
 
-    Resource.create!(pages: pages,
-                     abbreviation: 'test',
-                     onesky_project_id: 1,
-                     name: name,
-                     description: description,
-                     resource_attributes: attributes,
-                     resource_type_id: 1,
-                     system_id: 1)
+    resource = Resource.create!(abbreviation: 'test',
+                                onesky_project_id: 1,
+                                name: name,
+                                description: description,
+                                resource_attributes: attributes,
+                                resource_type_id: 1,
+                                system_id: 1)
+    resource.pages = pages
```

Comment:
Why did you need to do this?

Reply:
it was some sort of weird initialization cycle bug. when the resource xml validation rule ran the resource property on the page was nil.

By assigning the pages afterwards the resource property is correctly set

Reply:
I'm guessing having the closure defined for pages causes it to be handled slightly differently within AR.